src/index.js

```diff
@@ -522,9 +518,37 @@ export default class HyperScreen extends React.Component {
       });
 
     if (delay) {
-      later(parseInt(delay, 10)).then(fetchPromise);
+      /**
+       * Delayed behaviors will only trigger after a given amount of time.
+       * During that time, the DOM may change and the triggering element may no longer
+       * be in the document. When that happens, we don't want to trigger the behavior after the time
+       * elapses. To track this, we store the timeout id (generated by setTimeout) on the triggering
+       * element, and then look it up in the document after the elapsed time. If the timeout id is not
+       * present, we update the indicators but don't execute the behavior.
+       */
+      var delayMs = parseInt(delay, 10);
+      var timeoutId = null;
+      timeoutId = setTimeout(() => {
+        // Check the current doc for an element with the same timeout ID
+        const timeoutElement = getElementByTimeoutId(this.state.doc, timeoutId.toString());
```

Comment:
Would it make sense to reuse the `id` attribute / set it with a generated value when not present? This might help removing the need for adding `visitNode` and `getElementByTimeoutId`, as you'd be able to simply use `getElementById`

Reply:
I considered using `id`, but decided it was cleaner and safer to keep HV-specific state outside of the user-defined attributes. For example, it's possible our generated ID would conflict with an ID elsewhere in the XML.